crates/block-producer/src/store/mod.rs

```diff
@@ -126,15 +130,21 @@ type InnerClient = store_client::ApiClient<InterceptedService<Channel, OtelInter
 /// Interface to the store's gRPC API.
 ///
 /// Essentially just a thin wrapper around the generated gRPC client which improves type safety.
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub struct StoreClient {
     inner: InnerClient,
 }
 
 impl StoreClient {
-    /// TODO: this should probably take store connection string and create a connection internally
-    pub fn new(store: InnerClient) -> Self {
-        Self { inner: store }
+    pub fn new(store_address: SocketAddr) -> Result<Self, Error> {
+        let store = {
+            let store_url = format!("http://{store_address}");
+            let channel = tonic::transport::Endpoint::try_from(store_url)?.connect_lazy();
```

Comment:
imo we can unwrap on the error variant here because `store_url` is now guaranteed to be a correct url endpoint from the `format!` construction above